action/protocol/execution/evm/evm.go

```diff
@@ -46,6 +46,19 @@ var (
 	ErrInconsistentNonce = errors.New("Nonce is not identical to executor nonce")
 )
 
+type (
+	// GetBlockHash gets block hash by height
+	GetBlockHash func(uint64) (hash.Hash256, error)
+
+	// DepositGasWithSGD deposits gas with Sharing of Gas-fee with DApps
+	DepositGasWithSGD func(context.Context, protocol.StateManager, address.Address, *big.Int, *big.Int) (*action.TransactionLog, error)
+
+	// SGDRegistry is the interface for handling Sharing of Gas-fee with DApps
+	SGDRegistry interface {
```

Comment:
sGDRegistry?

Reply:
IIP-15 doc name it `SGDRegistry`, to the evm module, it is a registry to query the SGD status for a contract
inside chainservice, the var is named `sgdIndexer`

Reply:
why should it be public here?

Reply:
this means an interface/func outside of the EVM module that evm relies on, similar to `GetBlockHash` and `DepositGasWithSGD` being public